cc2538-bsl.py

```diff
@@ -97,6 +104,62 @@ def mdebug(level, message, attr='\n'):
 class CmdException(Exception):
     pass
 
+class FirmwareFile(object):
+    def __init__(self, path):
+        """
+        Read a firmware file and store its data ready for device programming.
+
+        This class will try to guess the file type if python-magic is available.
+
+        If python-magic indicates a plain text file, and if IntelHex is
+        available, then the file will be treated as one of Intel HEX format.
+
+        In all other cases, the file will be treated as a raw binary file.
+
+        In both cases, the file's contents are stored in bytes for subsequent
+        usage to program a device or to perform a crc check.
+
+        Parameters:
+            path -- A str with the path to the firmware file.
+
+        Attributes:
+            bytes: A bytearray with firmware contents ready to send to the device
+        """
+        self._crc32 = None
+
+        if firmware_auto_detect:
+            file_type = bytearray(magic.from_file(path, True))
+
+            #from_file() returns bytes with PY3, str with PY2. This comparison
+            #will be True in both cases"""
+            if file_type == b'text/plain':
+                mdebug(5, "Firmware file: Intel Hex")
+                self.bytes = bytearray(IntelHex(path).tobinarray())
+                return
+
+        else:
+            mdebug(5, "Cannot auto-detect firmware filetype: Assuming .bin")
+            mdebug(10, "You will not be able to programme your device with a " \
```

Comment:
Typo -> program? Not sure about this one. If it's correct British English, disregard this comment. ;)